lib/launcher/src/Cardano/Launcher.hs

```diff
@@ -250,16 +249,8 @@ launcherLogText MsgLauncherCleanup = "Terminating child process"
 -- other settings.
 --
 -- On Windows the current console code page is changed to UTF-8.
-setUtf8Encoding :: IO ()
-#if WINDOWS
-setUtf8Encoding = do
-    let utf8CodePage = 65001
-    setConsoleCP utf8CodePage
-    setConsoleOutputCP utf8CodePage
-    setUtf8EncodingHandles
-#else
-setUtf8Encoding = setUtf8EncodingHandles
-#endif
+withUtf8Encoding :: IO a -> IO a
+withUtf8Encoding action = withCP65001 (setUtf8EncodingHandles >> action)
```

Comment:
I guess here that we're relying on:

> Exports functions which adjust code pages on Windows, and **do nothing on other operating systems**.
(from http://hackage.haskell.org/package/code-page-0.2/docs/System-IO-CodePage.html)

Looking through the source of that package, it does seem that the above statement is probably true.

But would it be clearer here to differentiate between Windows and non-Windows, by only calling `withCP65001` if `WINDOWS` is defined?

Reply:
I think that the function haddock explains things enough.
Although CPP is a useful tool, I try to avoid it where possible, because it tends to break things. So I am happy there is a library function that has done the platform conditionals for us.